### DISCUSSION TITLE
default primary key 'ID' in Milvus DB exhausted

### DISCUSSION THREAD

**Discussion by anic1618** [milvus-io/milvus#32054] [Q&A and General discussion]
If the default primary key ID in Milvus DB is of type DataType.INT64, what happens when all IDs are exhausted? Are IDs reused when entities are deleted?

**Comment by yhmo** [comment]
The range of int64 is -2^63 ~ 2^63-1, I don't think any application can exhaust the ids.

**Comment by xiaofan-luan** [comment]
> If the default primary key ID in Milvus DB is of type DataType.INT64, what happens when all IDs are exhausted? Are IDs reused when entities are deleted?

there is no way  a INT64 field run out..
